driver/bpf/fillers.h

```diff
@@ -571,236 +571,158 @@ FILLER(sys_poll_x, true)
 	return bpf_poll_parse_fds(data, false);
 }
 
-#define MAX_IOVCNT 32
-#define MAX_IOVCNT_COMPAT 8
+#ifdef CONFIG_COMPAT
+	#define MAX_IOVCNT 8
+#else
+	#define MAX_IOVCNT 32
+#endif
 
-static __always_inline int bpf_parse_readv_writev_bufs_64(struct filler_data *data,
+static __always_inline int bpf_parse_readv_writev_bufs(struct filler_data *data,
```

Comment:
to be honest I would prefer to revert this code to what we had before this PR https://github.com/falcosecurity/libs/pull/1814/files#diff-a8b6fc07077f8fcef2f22e51392da619a5e68b7c603c62b3ab93ce5b6be22d6b. The new implementation seems pretty easy to break and the only advantage is that we support `readv` and `writev` on 32 bits... that IMO is a corner case.

Btw if you feel confident we can also merge it and maybe revert it if we see it doesn't work well...

Reply:
Yep, i would love to be able to just use `memcpy` there (the clang builtin of course); but it won't work :/ it would make the code much better...